src/cargo/ops/registry.rs

```diff
@@ -374,6 +382,61 @@ fn transmit(
     Ok(())
 }
 
+fn wait_for_publish(
+    config: &Config,
+    registry_src: SourceId,
+    pkg: &Package,
+    timeout: std::time::Duration,
+) -> CargoResult<()> {
+    let version_req = format!("={}", pkg.version());
+    let query = Dependency::parse(pkg.name(), Some(&version_req), registry_src)?;
+
+    let now = std::time::Instant::now();
+    let sleep_time = std::time::Duration::from_secs(1);
+    let mut logged = false;
+    loop {
+        let registry_description;
+        {
+            let mut registry = PackageRegistry::new(config)?;
+            let _lock = config.acquire_package_cache_lock()?;
+            registry.lock_patches();
```

Comment:
Wanted to call out that I'm recreating a `PackageRegistry` on each pass of the wait because I didn't see a way to clear it to force a new update.  Didn't see downsides to this (performance of it shouldn't be an issue with everything else going on) so I went with it

Reply:
Thanks! I overlooked that after getting burned with `cargo add` for not looking things up in the right layer